.github/workflows/ci.yml

```diff
@@ -0,0 +1,97 @@
+# Builds Betaflight Configurator on Windows, Android, Linux and macOS platforms.
+#
+# After building, artifacts are released to a seperate repository.
+
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+      - '*-maintenance'
+  pull_request:
+    branches:
+      - master
+      - '*-maintenance'
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version-file: '.nvmrc'
+          cache: yarn
+
+      - run: yarn install --immutable --immutable-cache --check-cache
+
+      - name: Run unit tests
+        run: yarn test
+
+      - name: Publish Unit Test Results
+        uses: EnricoMi/publish-unit-test-result-action@043296c976c53f4536194ebe3841ed720e04d496 # v1.26
+        if: always()
+        with:
+          files: test-results-junit/**/*.xml
+          comment_mode: 'off'
+
+  build:
+    name: Build (${{ matrix.name }})
+    needs: test
+    runs-on:  ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - name: Android
+            os: ubuntu-20.04
+            releaseArgs: --android
+
+          - name: Linux
+            os: ubuntu-20.04
+            releaseArgs: --linux64
+
+          - name: macOS
+            os: macos-11
+            releaseArgs: --osx64
+
+          - name: Windows
+            os: windows-2022
+            releaseArgs: --win64
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Cache NW.js
+        uses: actions/cache@v2
+        with:
+          path: cache/
+          key: ${{ runner.os }}-${{ hashFiles('cache/**') }}
+
+      - name: Install Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version-file: '.nvmrc'
+          cache: yarn
+
+      - name: Install Java JDK 8
+        uses: actions/setup-java@v2
+        if: ${{ matrix.name == 'Android' }}
+        with:
+          distribution: temurin
+          java-version: '8'
+
+      - run: yarn install --immutable --immutable-cache --check-cache
+
+      # - run: yarn gulp release ${{ matrix.releaseArgs }}
+      - run: yarn gulp debug-release ${{ matrix.releaseArgs }}
+
+      - name: Publish build artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: Betaflight-Configurator-${{ matrix.name }}
+          path: release/
+          retention-days: 3
```

Comment:
I suppose this retention days are only to store them in Github Actions? If we "release" a nightly for example in the betaflight-configurator-nightlies repo it does not affect, true?

Reply:
Good catch! I reduced this for testing, but it might be better to keep the pull request artifacts longer?
The artifacts uploaded to GitHub releases are not removed.

Reply:
If posible at least one week or a little more. If you find a way to attach or link to the artifacts in a response to the PR, we can't revisit this. But by now, I think 10-14 days is enough.

Reply:
Okay I updated it to 10 days.

Reply:
10 days is definite better than 7 days yes. Don't know about storage capacity we have.

Reply:
I think we don't have storage limit. GitHub let's store until 400 days I think (default 90 days).